lib/array_intersection.rb

```diff
@@ -1,6 +1,27 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n+m) where n represents the length the smaller array, and m the larger.
+# Space complexity: O(n), linear, because the additional space required is directly related to
+# length "n" of the smaller array.
 def intersection(array1, array2)
-  raise NotImplementedError
+  if array1 == nil || array2 == nil
+    return []
+  end
+  if array1.length < array2.length
```

Comment:
Since you are going to loop through each array it doesn't matter which one is smaller.